lib/src/disposable/disposable.dart

```diff
@@ -196,9 +197,13 @@ class Disposable implements _Disposable, DisposableManagerV3 {
     _throwOnInvalidCall('awaitBeforeDispose', 'future', future);
     _awaitableFutures.add(future);
     future.then((_) {
-      _awaitableFutures.remove(future);
+      if (!isDisposedOrDisposing) {
+        _awaitableFutures.remove(future);
+      }
     }).catchError((_) {
-      _awaitableFutures.remove(future);
+      if (!isDisposedOrDisposing) {
+        _awaitableFutures.remove(future);
+      }
```

Comment:
🔮  You could use `whenComplete` to handle the error and success case in one statement.